### ISSUE TITLE
Link `outputSourceBindings` to build

### ISSUE THREAD

**Issue by shashwathi** [knative/build-pipeline#173]
Below is a pipeline with a task which has an input source to `push-kritis` and outputs the image to `kritisImage` resource. Currently there is no mechanism to capture the `kritisImage` metadata. 
We need this information to use specific version of resource in further tasks of the pipeline.
 
```
apiVersion: pipeline.knative.dev/v1alpha1
kind: Pipeline
metadata:
  name: kritis-pipeline
  namespace: default
spec:
  tasks:
    - name: push-kritis              
      taskRef:
        name: build-push
      inputSourceBindings:
        - name: kritis-app-github
          key: workspace 
          resourceRef:
            name: kritis-repo-git
      outputSourceBindings:
        - name: kritisImage
          key: builtImage 
          resourceRef:
            name: kritis-resources-image
```

## Expected Behavior
Build or TaskRun should hold resource references to `outputSourceBindings` generated.

## Actual Behavior
`outputSourceBindings` of task is not used.

## Additional Info
This should be completed before tackling: https://github.com/knative/build-pipeline/issues/148
This change might require build to be rendered inside build-pipeline.

**Comment by bobcatfish** [comment]
@shashwathi could you provide some examples of what metadata you'd expect to see in Build or TaskRun?

**Comment by shashwathi** [comment]
@bobcatfish Sorry for late reply.

I started thinking in terms of metadata like this file

```
version: image-sha // incase of git probably commit sha
name: resource-name
```

Problem with this approach

1. After build is done, how do we access this? 
 _solution_: Probably volume or something that we can access after build completes.

2. Is there a world where task 1 uses github source `A` and alters the source but not make a commit? 
For example
```
task
name: create-task
source: git-source-A
steps:
- create-file
```
```
task:
name: push-task
source: git-source-A
passedConstraint: [create-task]
```

In this case step `push-task` expects the volume from `create-task` with the changes made. If this is a use case we need to support then we metadata is the wrong approach.
 
Next solution is providing volumes from `create-task` outputs that you can pass onto `push-task` as inputs. 
Does this make sense? Is there anything any use case I am missing?

**Comment by bobcatfish** [comment]
@shashwathi ah okay I think I understand now, thanks for the examples! - I think in fact that #124 might be a duplicate of this, would you agree?

**Comment by shashwathi** [comment]
I will close this issue. My bad I did not notice other issue

**Comment by bobcatfish** [comment]
>  My bad I did not notice other issue

no problem at all! better 2 issues than none :D
